smoke-test/src/test/java/org/opennms/smoketest/OpenNMSCloudPluginIT.java

```diff
@@ -0,0 +1,161 @@
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.smoketest;
+
+import static com.jayway.awaitility.Awaitility.await;
+import static java.lang.String.format;
+import static java.util.concurrent.TimeUnit.MINUTES;
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.time.Duration;
+
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.opennms.smoketest.stacks.OpenNMSStack;
+import org.opennms.smoketest.stacks.StackModel;
+import org.opennms.smoketest.utils.KarafShellUtils;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.utility.DockerImageName;
+import org.testcontainers.utility.MountableFile;
+
+import com.github.dockerjava.api.command.CreateContainerCmd;
+
+public class OpenNMSCloudPluginIT {
+
+    private static final String PATH_TO_FILE = "target/plugin-cloud-assembly/org.opennms.plugins.cloud-assembly.kar";
+    private static final String PATH_TO_CLOUD_MOCK = "target/cloud-mock-with-dependencies/org.opennms.plugins.cloud-mock-with-dependencies.jar";
+    private static final String CONTAINER_PATH = "/usr/share/opennms/deploy/";
+    private static final String CLOUD_MOCK_JAR_NAME = "org.opennms.plugins.cloud-mock-with-dependencies.jar";
+    private static final String CLOUD_MOCK_MAIN = "org.opennms.plugins.cloud.ittest.MockCloudMain";
+    private static final String KAR_FILE_NAME = "org.opennms.plugins.cloud-assembly.kar";
+
+    @ClassRule
+    public static OpenNMSStack stack = OpenNMSStack
+            .withModel(StackModel.newBuilder()
+                    .withSentinel()
+                    .build());
+
+    @ClassRule
+    public static GenericContainer mockCloudContainer = new GenericContainer(DockerImageName.parse("opennms/deploy-base:jre-2.0.6.b165"))
+            .withCreateContainerCmdModifier(cmd -> {
+                final CreateContainerCmd createCmd = (CreateContainerCmd) cmd;
+                createCmd.withEntrypoint("/bin/bash");
+                createCmd.withName("cloudMock");
+            }).withCopyFileToContainer(MountableFile.forHostPath(PATH_TO_CLOUD_MOCK), format("%s%s", CONTAINER_PATH, CLOUD_MOCK_JAR_NAME))
+            .withCommand("-c", format("java -cp %s%s %s", CONTAINER_PATH, CLOUD_MOCK_JAR_NAME, CLOUD_MOCK_MAIN)).withExposedPorts(9003)
+            .withNetwork(stack.opennms().getNetwork())
+            .withNetworkAliases("cloudMock");
+
+    @Before
+    public void setUp() {
+        mockCloudContainer.start();
+    }
+
+    @Test
+    public void installCloudPlugin() throws Exception {
+        installCloudPluginInOpennmsMustBeSuccessful();
+        installCloudPluginInSentinelMustBeSuccessful();
```

Comment:
Can we make this into two distinct tests by moving the `@Test` to the methods below?

Reply:
Actually not, because the sentinel installation will not work without having the opennms one done, that's why we put the two methods in one single test

Reply:
That is not possible. The sentinel plugin requires a successful install of the core plugin first.
Reason: The core plugin does the configuration and puts the certificates into the database via KV Store.
The Sentinel Plugin needs those certificates. The reason it was implemented that way:
The user needs to configure only one plugin.

Reply:
Ok, then this is fine. But it deserves a comment describing this dependency.

Reply:
ok